src/Microsoft.ML.AutoML/TransformInference/TransformInference.cs

```diff
@@ -420,6 +446,10 @@ public static SuggestedTransform[] InferTransforms(MLContext context, TaskKind t
             var labelColumnName = intermediateCols.FirstOrDefault(c => c.Purpose == ColumnPurpose.Label)?.ColumnName;
             concatColNames.Remove(labelColumnName);
 
+            // remove column with 'GroupId' purpose
+            var groupColumnName = intermediateCols.FirstOrDefault(c => c.Purpose == ColumnPurpose.GroupId)?.ColumnName;
+            concatColNames.RemoveAll(s => s == groupColumnName);
```

Comment:
You also need to remove "GroupId" too because it's a key column, it can't be cocat.

Reply:
I moved the custom GroupId column change to another PR. That's why there's only groupColumnName removed here.